src/components/ec-radio-btn/ec-radio-btn.vue

```diff
@@ -0,0 +1,358 @@
+            }"
+            :title="isTextInline ? radio.label : undefined"
+            :data-test="`ec-radio-btn__radio-label ec-radio-btn__radio-label-${radioIndex}`"
+          >
+            {{ radio.label }}
+          </label>
+
+          <p
+            v-if="radio.description"
+            :for="id"
+            class="ec-radio-btn__radio-description"
+            :class="{
+              'ec-radio-btn__radio-description--is-single-line': isTextInline,
+              'ec-radio-btn__radio-description--disabled': isDisabled,
+            }"
+            :title="isTextInline ? radio.description : undefined"
+            :data-test="`ec-radio-btn__radio-description ec-radio-btn__radio-description-${radioIndex}`"
+          >
+            {{ radio.description }}
+          </p>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div
+    :id="id"
+    v-if="isInvalid"
+    class="ec-radio-btn__error-text"
+    data-test="ec-radio-btn__error-text"
+  >
+    <slot name="error-message">{{ errorMessage }}</slot>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { StyleValue } from 'vue';
+import {
+  computed, ref, useAttrs, useSlots, withDefaults,
+} from 'vue';
+
+import { getUid } from '../../utils/uid';
+import EcIcon from '../ec-icon';
+import type { RadioBtnEvents, RadioBtnOption } from './types';
+import { RadioBtnEvent } from './types';
+
+interface RadioBtn extends RadioBtnOption {
+    isFocused: boolean
+}
+
+export interface RadioBtnProps {
+  options: RadioBtnOption[],
+  modelValue?: string,
+  label?: string,
+  errorMessage?: string,
+  isDisabled?: boolean,
+  isGroupInline?: boolean,
+  isTextInline?: boolean
+}
+
+const props = withDefaults(defineProps<RadioBtnProps>(), {
+  options: () => [],
+  modelValue: '',
+  label: '',
+  errorMessage: '',
+  isDisabled: false,
+  isGroupInline: false,
+  isTextInline: false,
+});
+
+const uid = getUid();
+const slots = useSlots();
+const attrs = useAttrs();
+const style = attrs.style as unknown as StyleValue;
+
+const id = `ec-radio-btn-${uid}`;
+const isInvalid = computed(() => (!!props.errorMessage || !!slots['error-message']));
+const errorId = computed(() => (isInvalid.value ? `ec-radio-btn-error-${uid}` : ''));
+
+const emit = defineEmits<{(e: 'update:modelValue', value: RadioBtnEvents[RadioBtnEvent.UPDATE_MODEL_VALUE]): void,
+}>();
+
+const radioButtons = ref<RadioBtn[]>([]);
+
+function addFocusPropertyToRadios(options: RadioBtnOption[]) {
+  radioButtons.value = options.map(radio => ({
+    ...radio,
+    isFocused: false,
+  }));
+}
+
+function isOptionChecked(radioValue: string) {
+  return props.modelValue === radioValue;
+}
+
+function isOptionFocused(radioIndex: number) {
+  return radioButtons.value[radioIndex].isFocused;
+}
+
+function setFocus(radioIndex: number) {
+  radioButtons.value[radioIndex].isFocused = true;
+}
+
+function unsetFocus(radioIndex: number) {
+  radioButtons.value[radioIndex].isFocused = false;
+}
```

Comment:
why don't we consider to create function `setFocusState(radioIndex: number, state: boolean)`?